Homework3/Task2Stack/Stack.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2Stack
+{
+    public class Stack<T> : IStack<T>
+    {
+        private readonly T[] myStack;
+        private int top;
+        private readonly int stackSize;
+
+        public Stack(int size)
+        {
+            stackSize = size > 0 ? size : 1;
+            top = -1;
+            myStack = new T[size];
+        }
+
+        public bool IsEmpty()
+        {
+            return top < 0 ;
+        }
+
+        public T Peek()
+        {
+            if(top < 0)
+            {
+                throw new IndexOutOfRangeException();
+            }
+            else
+            {
+                T value = myStack[top];
+                return value;
+            }
+        }
+
+        public T Pop()
+        {
+            if(top < 0)
+            {
+                throw new IndexOutOfRangeException();
+            }
+            else
+            {
+                T value = myStack[top--];
+                return value;
+            }
+
+        }
+
+        public void Push(T item)
+        {
+            if(top >= (stackSize-1))
+            {
+                throw new IndexOutOfRangeException();
+            }
+            else
+            {
+                myStack[++top] = item;
+            }
+            
+        }
+
+        public override string ToString()
+        {
+            StringBuilder tempString = new();
+
+            for (int i = top; i >= 0; i--)
+            {
+                tempString.Append($"{myStack[i]}, ");
+
+            }
+            
+            return tempString.ToString();
+        }
+
+        public Stack<T> Reverse()
```

Comment:
Is was not asked to add Reverse into existing class, but implement an extension method